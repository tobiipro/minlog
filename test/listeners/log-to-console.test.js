import MinLog from '../../src/minlog';
import _ from 'lodash';

import {
  _levelToConsoleFun
} from '../../src/listeners/log-to-console';

describe('logToConsole listener', function() {
  describe('_levelToConsoleFun', function() {
    it('should not regress (default levels)', function() {
      let instance = new MinLog();
      let snapshot = {};

      _.forEach(instance.levels, function(level, levelName) {
        let levels = [
          level - 1,
          level,
          level + 1,
          levelName,
          `x${levelName}x`
        ];

        _.forEach(levels, function(level) {
          snapshot[level] = _levelToConsoleFun({
            level,
            levels: instance.levels
          });
        });
      });

      expect(snapshot).toMatchSnapshot();
    });
  });
});
