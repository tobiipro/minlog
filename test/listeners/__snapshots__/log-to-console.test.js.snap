// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`logToConsole listener _levelToConsoleFun should not regress (default levels) 1`] = `
Object {
  "-1": "log",
  "0": "error",
  "1": "error",
  "10": "error",
  "11": "error",
  "19": "error",
  "20": "error",
  "21": "error",
  "29": "error",
  "30": "error",
  "31": "error",
  "39": "error",
  "40": "warn",
  "41": "warn",
  "49": "warn",
  "50": "warn",
  "51": "warn",
  "59": "warn",
  "60": "info",
  "61": "info",
  "69": "info",
  "70": "log",
  "71": "log",
  "79": "log",
  "80": "log",
  "81": "log",
  "89": "log",
  "9": "error",
  "90": "trace",
  "91": "log",
  "alert": "error",
  "critical": "error",
  "debug": "log",
  "emergency": "error",
  "error": "error",
  "fatal": "error",
  "info": "info",
  "informational": "info",
  "notice": "warn",
  "silly": "log",
  "trace": "trace",
  "verbose": "log",
  "warn": "warn",
  "warning": "warn",
  "xalertx": "log",
  "xcriticalx": "log",
  "xdebugx": "log",
  "xemergencyx": "log",
  "xerrorx": "log",
  "xfatalx": "log",
  "xinformationalx": "log",
  "xinfox": "log",
  "xnoticex": "log",
  "xsillyx": "log",
  "xtracex": "log",
  "xverbosex": "log",
  "xwarningx": "log",
  "xwarnx": "log",
}
`;
